input/postgres/collection_test.go

```diff
@@ -0,0 +1,92 @@
+package postgres_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/pganalyze/collector/input/postgres"
+	"github.com/pganalyze/collector/state"
+	"github.com/pganalyze/collector/util"
+)
+
+type collectionHelperTestpair struct {
+	Name               string
+	InputTypes         []string
+	ExpectedExists     bool
+	ExpectedReturnType string
+}
+
+var collectionHelperTests = []collectionHelperTestpair{
+	{
+		"explain_analyze",
+		[]string{"text", "text[]", "text[]", "text[]"},
+		true,
+		"text",
+	},
+	{
+		"get_stat_statements",
+		[]string{"boolean"},
+		true,
+		"SETOF pg_stat_statements",
+	},
+	// Verify we don't match on a shorter set of arguments
+	{
+		"explain_analyze",
+		[]string{"text", "text[]", "text[]"},
+		false,
+		"",
+	},
+	// Verify we don't match on a shorter set of arguments
+	{
+		"explain_analyze",
+		[]string{"text", "text[]", "text[]", "text[]", "text[]"},
+		false,
+		"",
+	},
+	// Verify we don't match on different argument types
+	{
+		"explain_analyze",
+		[]string{"text", "text[]", "text[]", "float"},
+		false,
+		"",
+	},
+	// Verify we allow default arguments to be ommitted
+	{
+		"get_stat_statements",
+		[]string{},
+		true,
+		"SETOF pg_stat_statements",
+	},
+}
+
+func TestCollectionFindHelper(t *testing.T) {
+	db := setupTest(t)
```

Comment:
Note this is defined in the [explain_analyze_test.go](https://github.com/pganalyze/collector/blob/main/input/postgres/explain_analyze_test.go) file.